pyqtgraph/graphicsItems/GradientLegend.py

```diff
@@ -45,14 +46,17 @@ def setLabels(self, l):
         
     def paint(self, p, opt, widget):
         UIGraphicsItem.paint(self, p, opt, widget)
-        rect = self.boundingRect()   ## Boundaries of visible area in scene coords.
-        unit = self.pixelSize()       ## Size of one view pixel in scene coords.
-        if unit[0] is None:  
-            return
-        
-        ## Have to scale painter so that text and gradients are correct size and not upside down
-        p.scale(unit[0], -unit[1])
 
+        p.save() # save painter state before we change transformation
+        view = self.getViewBox()
+        if view is not None:
+            trans = view.sceneTransform()
+            p.setTransform( trans ) # draw in ViewBox pixel coordinates
+            rect = view.rect()
+        else:
+            rect = QtCore.QRectF(0,0, 100, 100) # use dummy values when no associated ViewBox exists.
```

Comment:
probably should get rid of this line?

Reply:
Dear @j9ac9k , that's a good point.
When I added that, I thought there would still be some productive operations happening later in the code, but after cleaning up the transformations, that isn't really the case.
So the solution is probably to simply return from paint call early when there is no view box. With the dummy values plugged in, nothing useful would end up drawing anyway.

I changed to code to that approach now.